nautobot/core/api/views.py

```diff
@@ -552,3 +553,7 @@ def execute_graphql_request(self, request, data, query, variables, operation_nam
             return document.execute(**options)
         except Exception as e:
             return ExecutionResult(errors=[e], invalid=True)
+
+
+def json404(request):
```

Comment:
Sorry to come in at the 11th hour here, but I'd like to see this expanded to be more of `HttpOrJson404` pattern.

I think a better solution would be:

- Define [`handler404`](https://docs.djangoproject.com/en/4.0/topics/http/views/#customizing-error-views) inside of `nautobot.core.urls` to point to the custom view defined here.
- Inside of the custom view, inspect `request.path`:
  - If it starts with `/api/` raise a [`rest_framwork.exceptions.NotFound` from DRF](https://www.django-rest-framework.org/api-guide/exceptions/#notfound) which returns a JSON 404 response.
   - Else return the [default `django.views.defaults.page_not_found` 404 view from Django](https://docs.djangoproject.com/en/4.0/ref/views/#the-404-page-not-found-view) which returns a HTTP 404 response.

Reply:
The reason for this is it leaves the HTTP 404 responses the same as they are now, and makes sure that any JSON 404 responses are identical to ones that aren't customized.

Reply:
Good idea! Will do now...

Reply:
I modified the code as suggested, but found that raising NotFound actually returns a 500 response (weirdly) so stuck with a JsonResponse.